geps/gep-1619/index.md

```diff
@@ -604,8 +618,8 @@ and/or [GEP-2648](/geps/gep-2648) for more specific details on how to handle ove
 Edge cases will arise when implementing session persistence support for both backends and route rules through
 `BackendLBPolicy` and the route rule's `sessionPersistence` field. For guidance on addressing conflicting
 attachments, please consult the [Edge Case Behavior](#edge-case-behavior) section, which outlines API
-use cases. Only a subset of implementations have already designed their data plane to incorporate route-level session
-persistence, making it likely that route-level session persistence will be less widely implemented.
+use cases. Only a subset of implementations have already designed their data plane to incorporate route rule level session
+persistence, making it likely that route rule level session persistence will be less widely implemented.
 
 ### Traffic Splitting
 
```

Comment:
Since we dropped the `weight 0` scenario, I think we should still place a call out for `weight 0`, as it could be a ambiguous situation. I think the `Traffic Splitting` section makes the most sense. You can use own judgement if you disagree. I can't suggest for this line, so commenting this nearby:

```
When a
persistent session is established and traffic splitting is configured across services, the persistence to a single
backend MUST be maintained across services, **even if the weight is set to 0.**
```

Reply:
Thanks for pointing out.  I forgot to add more context about why I removed the weight 0 example and also want to confirm I'm not missing anything critical.

```
1. Curl to /a which establishes a persistent session with servicev1
2. Curl to /b routes to servicev1 due to route persistence despite weight: 0 configuration
```
The previous example we had is doing two route rules with path `/a` and `/b`, and path `/b' has traffic split with weight 0.  Since each route rule should have its own persistent session, so applying cookies from persistent session `/a` to servicev1 should not affact `/b` traffic routing with servicev1.  As a result, `/b` should be routed to servicev2 with `weight: 100`.  Does it make sense?

Then, I was thinking about how a persistent session can be established with a `weight: 0` backend?  Maybe servicev1 had a non-zero weight and a persistent session established, and then route resource is re-deployed and servicev1 has a new `weight: 0` configuration, but I am not sure if we expect session persistence to work after deployment, so I stopped adding an example with `weight: 0`.  Is there other example with `weight: 0` that we can include?

Reply:
> Since each route rule should have its own persistent session, so applying cookies from persistent session /a to servicev1 should not affact /b traffic routing with servicev1. 

So, to be clear, even if the session persistence is attached to the Service via BackendLBPolicy, each route rule should have different cookies / persistent sessions? Hmm. I was originally thinking that the cookie would be associated with the service when attached via BackendLBPolicy, so that it could be shared among route rules (in the weight 0 example).

If it isn't shared among route rules when using BackendLBPolicy, would that be problematic for setting the cookie name? For instance, ServiceA with cookie named "foo". There's two route rules for /x and /y, but both go just to ServiceA. The cookie "foo" will encode the backendID for ServiceA. Won't the cookie be re-used for /x and /y (two route paths)?

I think regardless, you bring up a good point that I think we should clarify in the GEP: does attaching to Service mean cookies are reused across route rules?

Reply:
I think attaching BackendLBPolicy to a Service should NOT mean cookies are resued across route rules referenceing to the same service.  Let me update PR with a clarification on this.

Reply:
Another commit was pushed for this.  PTAL @gcs278

Reply:
Thanks for the clarification @ginayeh. LGTM